### ISSUE TITLE
Interceptor for Firefox

### ISSUE THREAD

**Issue by avraham0** [a85/POSTMan-Chrome-Extension#9021]
Are you planning to create an Interceptor Firefox add-on anytime soon?

**Comment by DannyDainton** [comment]
I don't believe this is something that we're planning to do but if there is enough interest in a FF interceptor, we might need to rethink this decision. 😁

**Comment by jeitnier** [comment]
+1 for adding support in Firefox. Edge might be good too since it is gaining popularity.

**Comment by dphov** [comment]
+1

**Comment by bernardovieirarocha** [comment]
Please an  Interceptor Firefox would be great since chrome sucks

**Comment by shashankawasthi88** [comment]
@upwardweb we are actively looking at the interceptor for Firefox will be updating this thread as and when we have a solid timeline.

**Comment by bhagatparwinder** [comment]
I support this mission 🚀

**Comment by devmrfitz** [comment]
Any updates?

**Comment by abhijitkane** [comment]
@devmrfitz Would Postman's proxy feature be helpful for your use case? We recently announced support for HSTS-enabled endpoints (https://blog.postman.com/postmans-proxy-now-fully-supports-https-endpoints/), so all request capturing should work.
Capturing the request will also include cookies that your browser included, so you should be able to use your browser's authenticated session too.

**Comment by devmrfitz** [comment]
Yes. That's what I'm currently using. GET was intercepted but POST wasn't. I'll try to figure out why

**Comment by GayanSandaruwan** [comment]
It would be great to have this for Firefox. I don't even remember the last time I used Chrome for development purposes.

**Comment by parulgarg123** [comment]
+1 for request on this feature

**Comment by shashankawasthi88** [comment]
@parulgarg123 @GayanSandaruwan can you try using Postman Proxy instead (https://learning.postman.com/docs/sending-requests/capturing-request-data/capturing-http-requests/). As Abhijit mentioned, we have added the support for HSTS enabled endpoints. Can you let us know if any of your use cases are not possible via Proxy?

**Comment by parulgarg123** [comment]
@shashankawasthi88 It's not a clean solutions as compared to Interceptor also need to configure multiple things, as a developer If I am using third party things for my work then I expect that they will work on fly with 0 configuration. Kinda Plug and Play

**Comment by kiyokodyele** [comment]
+1 coz chrome is not developer friendly than firefox.

**Comment by kiyokodyele** [comment]
> @parulgarg123 @GayanSandaruwan can you try using Postman Proxy instead (https://learning.postman.com/docs/sending-requests/capturing-request-data/capturing-http-requests/). As Abhijit mentioned, we have added the support for HSTS enabled endpoints. Can you let us know if any of your use cases are not possible via Proxy?

the interceptor has big advantage than the proxy option, it capture everything. as @parulgarg123 mention, it works right out of the box after installing. I tried using proxy in Firefox but I have to deal with non secure https connection errors. Even if I manage to install the certificates required to capture https, I don't think it would work to the version I'm using which is 7. I'm not using the latest version because the new UI is confusing.

**Comment by luisabellan** [comment]
+1 for request on this feature

**Comment by Gibbz** [comment]
Yes please!

**Comment by ymsaadi** [comment]
Please do it

**Comment by jrcii** [comment]
+1

**Comment by Boffice** [comment]
+1

**Comment by bengaudiosi-toast** [comment]
Please do it!

**Comment by brandonmlusk** [comment]
would also love to see support for Firefox!

**Comment by NikoFarrelly** [comment]
+1

**Comment by shashankawasthi88** [comment]
@NikoFarrelly @brandonmlusk @bengaudiosi-toast @Boffice - [Proxy](https://learning.postman.com/docs/sending-requests/capturing-request-data/capturing-http-requests/) should be able to capture the requests from Firefox. Any particular reason why we prefer Interceptor over proxy or is it just the ease of configuration?

**Comment by Boffice** [comment]
@shashankawasthi88 Proxy is an amazing tool for use cases were you need to work on request sent by various apps, but in my case i need browser traffic only and : 

> is it just the ease of configuration?

⌃
 That, and also:

1. There is no way to filter browser traffic, "URL must contain" is not good idea when i want to check all the traffic sent from browser.
2. I simply do not need all the traffic to be routed to postman.
3. I have to restructure my network and firewall to make it work. 

To summarize - biggest issue is my network and yes i can use chrome but yesterday when i was debugging i used chrome to send packets to postman and Firefox for dev tools, it was not very convenient.  to be more precise i have chrome only for postman interseptor.

**Comment by shashankawasthi88** [comment]
> @shashankawasthi88 Proxy is an amazing tool for use cases were you need to work on request sent by various apps, but in my case i need browser traffic only and :
> 
> > is it just the ease of configuration?
> 
> ⌃ That, and also:
> 
> 1. There is no way to filter browser traffic, "URL must contain" is not good idea when i want to check all the traffic sent from browser.
> 2. I simply do not need all the traffic to be routed to postman.
> 3. I have to restructure my network and firewall to make it work.
> 
> To summarize - biggest issue is my network and yes i can use chrome but yesterday when i was debugging i used chrome to send packets to postman and Firefox for dev tools, it was not very convenient. to be more precise i have chrome only for postman interseptor.

Hi @Boffice thanks for a detailed reply. While we address the issues that you pointed out, there is a way for you to get only the browser traffic in Postman via proxy. 
Firefox allows you to set a proxy host/ port just at the browser level, that ways only the browser traffic goes through Postman and not the traffic from other applications in the system.

**Comment by twinsant** [comment]
Why not?

**Comment by minlaxz** [comment]
+1 here, also for Edge.

**Comment by fennaneyahia** [comment]
+1 for Firefox. it would be amazing.

**Comment by abhijitkane** [comment]
@fennaneyahia @minlaxz @twinsant Thanks for reaching out - if you haven't already, the proxy (https://learning.postman.com/docs/sending-requests/capturing-request-data/capturing-http-requests/#step-1-enable-the-proxy) supports all of the use-cases that the Interceptor supported :). Addons like https://addons.mozilla.org/en-US/firefox/addon/switchyomega/ let you configure the proxy only for selected hosts, enabling selective filtering if required.

Do let us know if you have use-cases that the proxy feature is unable to fulfil.

**Comment by pzelnip** [comment]
> @fennaneyahia @minlaxz @twinsant Thanks for reaching out - if you haven't already, the proxy (https://learning.postman.com/docs/sending-requests/capturing-request-data/capturing-http-requests/#step-1-enable-the-proxy) supports all of the use-cases that the Interceptor supported :). Addons like https://addons.mozilla.org/en-US/firefox/addon/switchyomega/ let you configure the proxy only for selected hosts, enabling selective filtering if required.
> 
> Do let us know if you have use-cases that the proxy feature is unable to fulfil.

How about if I google "Postman Interceptor" I end up landing on this page: https://learning.postman.com/docs/sending-requests/capturing-request-data/interceptor/ none of which helps if one is using completely different extensions for the use cases that Interceptor covers.

Sure you might be able to achieve the same results using different tools, but that's a whole lot of figuring out I'd like to avoid.

**Comment by zesty** [comment]
Proxy for intercepting traffic doesn't work for me on linux with firefox for anything but plain http. I've installed the postman cert in firefox with website trust, and disable cert verification in postman settings. I've tried using both foxyproxy and firefox's own manual proxy config. Tried a blank, refreshed firefox profile with zero addons. Just. Doesn't. Work.

mostly I get:

Secure Connection Failed
An error occurred during a connection to example.com.
    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the website owners to inform them of this problem.

if I use https/ssl type in foxyproxy I can get: SSL_ERROR_RX_RECORD_TOO_LONG

Interceptor in chrome worked fine.

On the flip side, ***I did finally manage to get postman to use ZAP as a proxy. The secret sauce was to disable ALPN in the ZAP proxy settings, otherwise postman would show "Error: socket hang up"*** even though ZAP handled the request fine. I made an additional local proxy in ZAP on port 8081 so as not to mess with my default proxy.

**Comment by SenuraMalaka** [comment]
+1 for Firefox.

**Comment by Shacobushi** [comment]
+ 1 Firefox, PLZZZ

**Comment by bc183** [comment]
+1 for Firefox

**Comment by TopengDev** [comment]
+1 FOR FIREFOX, NEEDED IT SO BAD

**Comment by akshaydeo** [comment]
👋 thank you so much for reporting this issue and showing interest in having an interceptor extension for Firefox.

We have some good news for all; we are releasing the Firefox interceptor as a part of our Aug 2023 release 🎉 . I will update this thread with launch details in Aug.

**Comment by akshaydeo** [comment]
https://blog.postman.com/announcing-the-postman-interceptor-extension-for-all-browsers/ 

We are in the process of getting our firefox add-on approved, but you will be able to install it now :).

**Comment by nativeit** [comment]
I'm glad I found this thread, I saw the Postman Interceptor extension in the "Trending" section of the Mozilla Add-ons website. I clicked through to view the developer profile, and it showed had only been created last month, which made me suspicious (I'm only now switching to Firefox, and had not yet realized Postman is just now filling this gap). The extension had a little over 200 users, no reviews or ratings, and all of this made me concerned that this was potentially some kind of honeypot or fake. I went to Google and searched for recent mentions of "Postman + Firefox", but the first few variations on that theme did not reveal anything to verify things. I figured I could find pull requests or something here, and wound up here. I was very close to reporting your developer account (not that my report would have resulted in anything ultimately happening.

If someone at Postman could post something to the website/socials (or maybe boost it if you already have), so a Google search for "Postman + Firefox" will quickly discourage any more0 paranoid jerks like myself, who might be tempted to report your extension.

tldr; Super pleased to see I chose the right time to move over to Firefox, and will have this extension! I'm going to be your first review, I think, and I just want to encourage anyone else that's installed it to do the same.

Congrats on getting this done. :100:
